### ISSUE TITLE
Installation fails

### ISSUE THREAD

**Issue by xanathon** [misskey-dev/misskey#6868]
## 💡 Summary

When trying to install I get up to step 5, then the installation fails
## 🙂 Expected Behavior

yarn should run

## ☹️ Actual Behavior

yarn throws an error:

```
misskey@h2881594:~/misskey$ yarn
00h00m00s 0/0: : ERROR: There are no scenarios; must have at least one.
```

## 📝 Steps to Reproduce

enter yarn into command line 

## 📌 Environment

Ubuntu 18 lts

**Comment by acid-chicken** [comment]
It seems the installed yarn is old. Installation guide for yarn can be seen here: https://classic.yarnpkg.com/en/docs/install/#debian-stable

**Comment by xanathon** [comment]
I issued

`apt-get install yarn`

minutes before I tried to use it. And it still may be old?

**Comment by acid-chicken** [comment]
Yes, even when you installed it just now, it may be old if you didn't add vendor's (dl.yarnpkg.com) apt repository before installation because the default repositories of Ubuntu often have older packages.

**Comment by xanathon** [comment]
Thanks, I'll try

**Comment by xanathon** [comment]
Okay, I installed the latest yarn and node.js 14.x, but it still does not work:


```
make: g++: Command not found
make: *** [Release/obj.target/diskusage/src/main.o] Error 127
diskusage.target.mk:114: recipe for target 'Release/obj.target/diskusage/src/main.o' failed
make: Verzeichnis â/home/misskey/misskey/node_modules/diskusage/buildâ wird verlassen
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
gyp ERR! System Linux 4.15.0
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/misskey/misskey/node_modules/diskusage
gyp ERR! node -v v14.15.1

```

**Comment by acid-chicken** [comment]
Please try `sudo apt install build-essential`.

**Comment by xanathon** [comment]
Thanks again.

If I start as misskey user I get 

```
INFO *  [core boot]     Welcome to Misskey!
INFO *  [core boot]     Misskey v12.60.1
INFO *  [core boot env] NODE_ENV: production
INFO *  [core boot env] You do not have root privileges
INFO *  [core boot nodejs]      Version 14.15.1
DONE *  [core boot config]      Loaded
INFO *  [core boot db]  Connecting...
DONE *  [core boot db]  Connected: v10.15 (Ubuntu 10.15-0ubuntu0.18.04.1)
ERR  *  [core boot]     You need root privileges to listen on well-known port on Linux

```

If I change to a user with root privileges I get:


```
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /home/misskey/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/home/misskey/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent
```

I am close to giving up. I found a docker for misskey, but sadly no installation instructions ...

**Comment by SanMurakami** [comment]
> I am close to giving up. I found a docker for misskey, but sadly no installation instructions ...

Just create the `default.yml` from [this page](https://github.com/syuilo/misskey/blob/develop/.config/example.yml) and mount it in `/misskey/.config` and it should work perfectly.

**Comment by xanathon** [comment]
I did create that file. In the installation instructions it only says "edit it". But there is no information on **what exactly to edit**. I entered a domain and entered a postgres database user. After that I created the same user with the password I did set in default.yml in postgres. I uncommented the line for https

I think there should be some clarification in the installation document on **what extactly** to edit in the configuration file, because I have no clue what else I have to edit to get it to work.

**Comment by xanathon** [comment]
Okay, I will stop trying. This is not in a state that is user friendly enough to get it installed at the moment. I know my way around linux and have installed quite a number of other applications (including Hubzilla), but here the installation instructions seems lacking and unfinished. You will need to work on that if you want a bigger userbase.

Thanks anyway and good luck with the project. I liked the look and the features, but this just is way too much effort. I also fear that I would get similar time-consuming problems when an update is neccessary. So I will look for other Fediverse-compatible software.

**Comment by SanMurakami** [comment]
In my opinion, Misskey is much easier to build than Mastodon.

Especially when it comes to config settings, the comments out should give you a general idea of what to expect.

The minimum requirements to implement Misskey in Docker are as follows
- PostgreSQL 11
- Redis

If you are not sure, please continue this thread.

**Comment by SanMurakami** [comment]
The minimum required configuration items are as follows.

```
url: http://yourdomain

port: 80

db:
  host: db_hostname
  port: db_port (default: 5432)
  db: db_name
  user: db_user
  pass: db_password

redis:
  host: redis_hostname
  port: redis_port (default: 6379)

id: 'aid'
autoAdmin: true
```

**Comment by xanathon** [comment]
I did set a subdomain.

Since I uncommented https in the config I think the port needs to be 443?

The db settings are correct.

I have never worked with redis before, so I have no idea what to enter into the fields there. A quick glance at the redis webseite and documentation did not help.

And what does "aid" mean for id?

**Comment by SanMurakami** [comment]
Sorry for the delay in replying to you.

> Since I uncommented https in the config I think the port needs to be 443?

You need to read this.
https://github.com/syuilo/misskey/blob/6a76118710c3ccd669a4a7623637784bf87ccb22/.config/example.yml#L11-L52

> I have never worked with redis before, so I have no idea what to enter into the fields there. A quick glance at the redis webseite and documentation did not help.

Install Redis, or launch it in Docker. That's it.

By the way, in what way are you going to use Docker?
Pure? Docker Compose? Kubernetes?

> "aid" mean for id?

https://github.com/syuilo/misskey/blob/6a76118710c3ccd669a4a7623637784bf87ccb22/.config/example.yml#L95-L111

You say that there is "no documentation", but the config is also well documented in the comments out, and if you were able to set up Hubzilla, you should be able to read it and understand it.

Misskey is much easier to set up than other Fediverse applications, and if you can't install Misskey, you probably won't be able to set up any other applications either.

**Comment by xanathon** [comment]
Docker-compose.

Sorry, but you are wrong. Hubzilla was way, **WAY** easier to install than Misskey. I have installed multiple Fediverse flavours for testing and most of them are **a lot** easier to install than Misskey.

I gave up. This just does not work, takes way too much time and the documentation is severely lacking. Comments in a config file are no "documentation". All this needs to be explained thoroughly in the installation docs.

If you want to reach a broader audience with Misskey, the installation needs to be simplified and the documentation needs to be a lot more verbose.
